### ISSUE TITLE
deploy to android use onnxruntime

### ISSUE THREAD

**Issue by huyletran** [rhasspy/openWakeWord-cpp#1]
How to deploy to android use onnxruntime? Thanks

**Comment by huyletran** [comment]
Hello author please check my issue

**Comment by ObscurusGrassator** [comment]
Hi, you can try/test `https://github.com/ObscurusGrassator/openWakeWord-cpp/tree/AndroidImplementation` .

**Comment by chinalwb** [comment]
@ObscurusGrassator Awesome work for migrating it onto Android platform!

I checked out your code, please correct me if I was wrong, the solution works similarly to this cpp implementation, I mean, it starts the microphone and then detect the keyword in a continuous manner.

For my use case, and I'd assume it should be a common scenario, the detection would be more likely works as `on-demand`. Let's say, when user launches app, it starts detecting, and if the keyword has been detected, it stops for some period (maybe the app will be waiting for some further voice command), and after that period, it expects to be started again to detect the keyword.

Simply put, I tried to refactor the cpp into JNI (for Android) into 3 functions:
1. start - this starts the detect engine but it won't do any work until some data from microphone gets collected;
2. check(byte[] audioData) - pass the audio data from microphone in, and notify the listener in java world when keyword gets recognized;
3. stop - this stops the threads in the engine.

I have no much experience with C++ / JNI, would you mind helping improve the AndroidImplementation? Or maybe the AndroidImplementation already been done like this?

Thank you! @ObscurusGrassator

**Comment by ObscurusGrassator** [comment]
Yes, the original C++ functionality has not been changed. I just mapped the input to the android microphone, and the output to the android service.

Yep, that's exactly how my version works. My example service send `end_after_activation = true` to C++ part and it end after wake word tetection, and started after manualy calling the service (intent) again.

Yes.

I don't have much experience with Android, Java and C++ :D . I studied all this language just for the purposes of OpenWakeWord :D . If anyone can improve my code, feel free!
